pages/api/account/manage/event/[[...data]].js

```diff
@@ -115,6 +130,15 @@ export async function updateEventApi(username, id, updateEvent) {
     return { error };
   }
 
+  
+  // Add to Changelog
+  logChange({
+    userId, 
+    collection: "events", 
+    changesBefore: beforeUpdate, 
+    changesAfter: await getEventApi(username, id)
```

Comment:
Can 1 more field be added to the model, for the string `CREATE`, `DELETE`, `UPDATE`

Reply:
> Can 1 more field be added to the model, for the string `CREATE`, `DELETE`, `UPDATE` 

Hmm, Okay!

So the new field would be meant to track the CRUD Operation Type!?

`operation` good for a name!?

________

> - then can we add the `logChange` to the create and delete functions also?

We already have it integrated in the `create` endpoints for the respective models, do you mean something different when you say "functions"?

Also, I'd intentionally not integrated the `logChange` in the `delete` endpoints, because it's a deletion, is there a reason or a usecase for the deletion operation changelog!? I'd like a peek into your thoughts, I'm curious 🤓

Reply:
> operation good for a name!?

sounds good 👍 

---

> We already have it integrated in the create endpoints for the respective models, do you mean something different when you say "functions"?

For example In the events API file, the logchange is not on the function that does the create and delete, I think if we displayed a change log to admins or the user it would be super helpful

Simple example we might display

| Section | Date | Before | After | Action |
| :--- | :--- | :--- | :--- | :--- |
| Events | 12th September 2023 14:34 | - | `{...}` | Created |
| Events | 13th September 2023 14:34 | `{...}` | `{...}` | Updated |
| Events | 14th September 2023 14:34 | `{...}` | - | Deleted |

> Also, I'd intentionally not integrated the logChange in the delete endpoints, because it's a deletion, is there a reason or a usecase for the deletion operation changelog!? I'd like a peek into your thoughts, I'm curious 🤓

For example when I delete an event, it would be good to know the latest state of that db entry before it is deleted.

Reply:
> and the logChange integrations missing in the events endpoint function!? 😳 I must've missed that part oh wow

no I don't think you missed it, I think if the developer doesn't use http and calls the function directly with nextjs, then the userId won't get populated

Reply:
Oh, I knew that... I thought you said that the `logChange` function wasn't called/implemented in the `event` endpoint function like we did in the other endpoint